### PR TITLE
Fix null type issue

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

**Review by pakrym — APPROVED** [review]

**Review by kasobol-msft — APPROVED** [review]

File: sdk/core/System.Memory.Data/src/BinaryData.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public BinaryData(byte[] data)
         public BinaryData(object? jsonSerializable, JsonSerializerOptions? options = default, Type? type = default)
 #pragma warning restore AZC0014 // Avoid using banned types in public API
         {
-            _bytes = JsonSerializer.SerializeToUtf8Bytes(jsonSerializable, type, options);
+            _bytes = JsonSerializer.SerializeToUtf8Bytes(jsonSerializable, type ?? jsonSerializable?.GetType() ?? typeof(object), options);
```
Comment (KrzysztofCwalina) [inline sdk/core/System.Memory.Data/src/BinaryData.cs:55]: Do we want typeof(object) or jsonSerializable.GetType()? I think we want the later. typeof(object) is a pretty useless payload to serialize.

File: sdk/core/System.Memory.Data/tests/BinaryDataTests.cs [hunk 2]
```diff
@@ -299,6 +299,8 @@ public void CanCreateBinaryDataFromCustomType()
             AssertData(BinaryData.FromObjectAsJson(payload));
             AssertData(BinaryData.FromObjectAsJson(payload, new Text.Json.JsonSerializerOptions { IgnoreNullValues = true }));
             AssertData(new BinaryData(payload, type: typeof(TestModel)));
+            AssertData(new BinaryData(payload));
```
Comment (KrzysztofCwalina) [inline sdk/core/System.Memory.Data/tests/BinaryDataTests.cs:302]: We should have a test that passes instance of a subtype but Type representing the instance's base type. The test should verify that only the base properties are serialized.

**Review by KrzysztofCwalina — CHANGES_REQUESTED** [review]

File: sdk/core/System.Memory.Data/src/BinaryData.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/core/System.Memory.Data/src/BinaryData.cs:55]: Yes, we first try jsonSerializable.GetType:
``` 
type ?? jsonSerializable?.GetType() ?? typeof(object)
```

Comment (JoshLove-msft) [reply sdk/core/System.Memory.Data/src/BinaryData.cs:55]: Just to clarify - are you suggesting that we don't fall back to typeof(object) at all?

File: sdk/core/System.Memory.Data/src/System.Memory.Data.csproj [hunk 3]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <Description>Contains the BinaryData type, which is useful for converting between strings, streams, JSON, and bytes.</Description>
     <AssemblyTitle>System.Memory.Data</AssemblyTitle>
-    <Version>1.0.0</Version>
+    <Version>1.0.1-beta.1</Version>
```
Comment (AlexGhiondea) [inline sdk/core/System.Memory.Data/src/System.Memory.Data.csproj:5]: Do we need a beta.1 here? We usually just release the patch version.

Comment (JoshLove-msft) [reply sdk/core/System.Memory.Data/src/System.Memory.Data.csproj:5]: Yes, it was suggested that I add this just as an extra safeguard but as long as the release date is not set, there is no danger of it being released by accident. Will remove.

File: sdk/core/System.Memory.Data/src/BinaryData.cs (hunk 1 above)
Comment (KrzysztofCwalina) [reply sdk/core/System.Memory.Data/src/BinaryData.cs:55]: Yeah, I would not fall back to typeof(object).

**Review by KrzysztofCwalina — APPROVED** [review]

File: sdk/core/System.Memory.Data/src/BinaryData.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/core/System.Memory.Data/src/BinaryData.cs:55]: Also, as of .NET 5.0, null is not allowed to be passed here - https://github.com/dotnet/runtime/pull/44633#discussion_r522693231

Comment (ericstj) [reply sdk/core/System.Memory.Data/src/BinaryData.cs:55]: Right, so today when passing in `jsonSerializable: null, type: null` you get back `null` from the serializer.  In 5.0 this [throws](https://docs.microsoft.com/en-us/dotnet/core/compatibility/serialization#jsonserializerserialize-throws-argumentnullexception-when-type-parameter-is-null) ANE.  By falling back to passing `typeof(object)` here you'll have consistent behavior which is `null`.

**Pull request merged**